### PR TITLE
Initial Commit! 

### PR DESCRIPTION
**Pull request by brentru**
**Description**: This library enables access to Microsoft Azure IoT device, messaging, and job services from a CircuitPython wifi-enabled device.

**Azure IoT Library Scope (some of these features are being added and not in the initial PR)** 
* Device-to-Cloud messaging
* Cloud-to-Device messaging
* Hub Service (Device Management): Create, retrieve, update.
* Device Twin Management: Create, retrieve, update.
* HTTP Error Handling

### CODE REVIEW

**Comment by brentru** [comment]
This PR is ready for review!

**Comment by brentru** [comment]
Modified code, usage now dictates that you place the CircuitPython device's `device_id` within a `secrets.py` dictionary file instead of within a method call.

File: adafruit_azureiot.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        :param str status_reason: Description of HTTP status.
+        """
+        for error in AZURE_HTTP_ERROR_CODES:
+            if error == status_code:
+                raise TypeError("Error {0}: {1}".format(status_code, status_reason))
+
+    # Cloud-to-Device Messaging
+    def get_hub_message(self):
+        """Returns a message from a Microsoft Azure IoT Hub (Cloud-to-Device), or -1
+        if the message queue is empty.
```
Comment (tannewt) [inline adafruit_azureiot.py]: One tiny thing here would be to return None when empty rather than -1.

Comment (brentru) [reply adafruit_azureiot.py]: Good idea, added this in PR https://github.com/adafruit/Adafruit_CircuitPython_AzureIoT/pull/3

**Pull request merged**